publication-rest/src/main/java/no/unit/nva/publication/update/UpdatePublicationHandler.java

```diff
@@ -86,9 +92,40 @@ public UpdatePublicationHandler(ResourceService resourceService,
         this.uriRetriever = uriRetriever;
     }
 
+    private static boolean isAlreadyPublished(Publication existingPublication) {
+        return PublicationStatus.PUBLISHED.equals(existingPublication.getStatus())
+                || PublicationStatus.PUBLISHED_METADATA.equals(existingPublication.getStatus());
+    }
+
+    private static boolean isPending(TicketEntry publishingRequest) {
+        return TicketStatus.PENDING.equals(publishingRequest.getStatus());
+    }
+
+    private static boolean hasMatchingIdentifier(Publication publication, TicketEntry ticketEntry) {
+        return ticketEntry.getResourceIdentifier().equals(publication.getIdentifier());
+    }
+
+    private static URI getCustomerId(Publication publicationUpdate) {
+        return publicationUpdate.getPublisher().getId();
+    }
+
+    private boolean userIsContributor(RequestInfo requestInfo, Publication publication) {
```

Comment:
Not sure, but could this be more effective?
```
private boolean userIsContributor(RequestInfo requestInfo, Publication publication) {
        return publication.getEntityDescription()
                .getContributors()
                .stream()
                .filter(contributor -> Objects.nonNull(contributor.getIdentity()))
                .map(contributor -> contributor.getIdentity().getId())
                .anyMatch(cristinId -> attempt(() -> cristinId.equals(requestInfo.getPersonCristinId())).orElseThrow());
    }
```
Also, maybe add a nonNull check for publication.getEntityDescription().getContributors()?